src/io/output/FileOutputHandler.cpp

```diff
@@ -1,35 +1,50 @@
 #include "FileOutputHandler.h"
 
 #include <filesystem>
-#include <iostream>
 
 #include "io/logger/Logger.h"
 
-FileOutputHandler::FileOutputHandler(const OutputFormat output_format, const std::string& output_dir_path)
-    : output_format(output_format), output_dir_path(output_dir_path) {
-    switch (output_format) {
-        case OutputFormat::VTK:
-            file_writer = std::make_unique<VTKWriter>();
+FileOutputHandler::FileOutputHandler(const SimulationParams& params) : params(params) {
+    switch (params.output_format) {
+        case OutputFormat::NONE:
+            break;
+        case OutputFormat::VTU:
+            file_writer = std::make_unique<VTUWriter>();
             break;
         case OutputFormat::XYZ:
             file_writer = std::make_unique<XYZWriter>();
             break;
-        case OutputFormat::NONE:
-            return;
+        case OutputFormat::CHKPT:
+            file_writer = std::make_unique<CheckPointWriter>();
+            break;
         default:
             Logger::logger->error("Output format not implemented.");
             exit(1);
     }
 
-    if (std::filesystem::exists(output_dir_path)) {
-        std::filesystem::remove_all(output_dir_path);
+    if (std::filesystem::exists(params.output_dir_path)) {
+        auto supported = get_supported_output_formats();
+        auto file_extension = std::find_if(supported.begin(), supported.end(), [params](const auto& pair) {
+                                  return pair.second == params.output_format;
+                              })->first;
+
+        auto count = 0;
+        for (const auto& entry : std::filesystem::directory_iterator(params.output_dir_path)) {
+            if (entry.path().extension() == "." + file_extension) {
+                std::filesystem::remove(entry.path());
+                count++;
+            }
+        }
+
```

Comment:
I don't know what you wanted to do with this count but I suppose some logging?
Maybe something like in the suggestion.
```suggestion
Logger::logger->warn("Removed {} files with targetted file extension {} from target directory {}", count, file_extension, params.output_dir_path); 
```